lib/tx.py

```diff
@@ -78,6 +78,7 @@ class Deserializer(object):
     def __init__(self, binary, start=0):
         assert isinstance(binary, bytes)
         self.binary = binary
+        self.binary_length = len(binary)
```

Comment:
I would recommend leaving this change in. While `len()` is nice and explicit here, those timings show over 200% speedup for referencing the previous stored length over calling `len()` again. While its only shaving off tens of milliseconds, those add up over the many times the code is reached. Profiling before the change has it taking up almost 1% of my sync time: 
![image](https://user-images.githubusercontent.com/577312/29009869-9d8c36c2-7aec-11e7-8039-18cae11842e6.png)